### ISSUE TITLE
Edit and Continue silently fails if custom PathMap is defined

### ISSUE THREAD

**Issue by vsfeedback** [dotnet/roslyn#47194]
_This issue has been moved from [a ticket on Developer Community](https://developercommunity.visualstudio.com/idea/1164519/edit-and-continue-silently-fails-if-custom-pathmap.html)._

---
[severity:I'm somewhat annoyed. A fix would be nice]
This isn't technically a problem with Visual Studio itself but rather a consequence of configuring the project in such a way that causes Visual Studio to behave unexpectedly. If you define a custom PathMap property in the csproj, Edit and Continue will silently stop working while debugging. Visual Studio was also unhelpful in identifying the problem as it simply behaved as though Edit and Continue was turned off. I reiterate that this was an issue with the project's configuration; I moved PathMap to the release configuration, but I would greatly appreciate if Visual Studio tried to warn me if Edit and Continue isn't going to work.

---
#### Feedback Bot on 8/26/2020, 07:01 PM: 

We have converted this feedback item from problem to idea. This change was done to better reflect the feedback's nature.
#### Feedback Bot on 8/27/2020, 02:57 AM: 

Thank you for taking the time to provide your suggestion.&nbsp; We will do some preliminary checks to make sure we can proceed further.&nbsp; We'll provide an update once the issue has been triaged by the product team.

**Comment by jinujoseph** [comment]
cc @tmat

**Comment by tmat** [comment]
I agree we should report a warning that this msbuild property is set when you start debugging. 

Rather then setting this property manually I'd recommend using [Source Link](https://github.com/dotnet/sourcelink) and setting `ContinuousIntegrationBuild` property to true on your CI machine.

**Comment by lambdageek** [comment]
This also affects `WatchHotReloadService`.  `StartSessionAsync` succeeds, but every call to `EmitSolutionUpdateAsync` returns empty updates and empty diagnostics.
